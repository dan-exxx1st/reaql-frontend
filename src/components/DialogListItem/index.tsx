import React, { FC } from 'react';
import StyledDialogListItem from './style';

const DialogListItem: FC = () => {
	return <StyledDialogListItem>Dialog list item</StyledDialogListItem>;
};

export default DialogListItem;
