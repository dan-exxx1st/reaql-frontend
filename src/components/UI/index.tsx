export { default as Avatar } from './Avatar';
export { default as Button } from './Button';
export { default as Circle } from './Circle';
export { default as SkeletonLoader } from './SkeletonLoader';
export { default as TextField } from './TextField';
export { default as Typography } from './Typography';
export { default as Wrapper } from './Wrapper';
export { default as CheckMark } from './CheckMark';
export { default as Icon } from './Icon';
export { default as CheckBox } from './CheckBox';
export { default as SystemMessage } from './SystemMessage';
export { default as Spinner } from './Spinner';
