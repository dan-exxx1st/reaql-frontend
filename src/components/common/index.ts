export { default as DialogHeader } from './DialogHeader';
export { default as ContactItem } from './ContactsItem';
export { default as UserHeader } from './UserHeader';
export { default as DialogTextField } from './DialogTextField';
export { default as MessageItem } from './MessageItem';
export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';
export { default as MessageSearch } from './MessageSearch';
