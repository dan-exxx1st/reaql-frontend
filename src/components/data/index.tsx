export { default as MessageListWithData } from './MessageListWithData';
export { default as DialogHeaderWithData } from './DialogHeaderWithData';
