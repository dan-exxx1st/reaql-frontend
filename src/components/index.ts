export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
