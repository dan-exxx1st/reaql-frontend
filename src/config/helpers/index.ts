import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
