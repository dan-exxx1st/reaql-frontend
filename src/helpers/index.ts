import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
