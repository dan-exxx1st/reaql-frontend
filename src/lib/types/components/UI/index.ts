import { MouseEventHandler } from 'react';
import Theme from 'helpers/styled';
import { ColorType, IconName } from 'lib/types';
import { TypographyVariantType, TypographyComponentType } from 'lib/types/ui';

export interface IAvatarProps {
    src?: HTMLImageElement['src'];
    alt?: string;
    className?: string;
}

export interface IButtonProps {
    className?: string;
    theme: typeof Theme;
    onClick?: MouseEventHandler;
    variant: 'filled' | 'outlined';
    color: ColorType;
}

export interface ICheckMarkProps {
    isChecked: boolean;
    isDouble: boolean;
}

export interface ICircleProps {
    color: ColorType;
    iconName?: IconName;
    text?: string | number;
    width?: string;
    height?: string;
    cursor?: string;
    className?: string;
    onClick?: () => void;
}

export interface IIconProps {
    iconName: IconName;
    className?: string;
}

export interface ISkeletonLoaderProps {
    variant?: 'avatar' | 'block';
    width?: string;
    height?: string;
    className?: string;
}

export interface ITextFieldProps {
    onChange?: (value: string) => void;
    value?: string;
    className?: string;
    fieldSize: 'small' | 'medium' | 'large';
    color: ColorType;
    icon?: IconName;
    width?: string;
    height?: string;
    wrapperWidth?: string;
    placeholder?: string;
    type?: string;
}

export interface ITypographyProps {
    className?: string;
    variant: TypographyVariantType;
    component?: TypographyComponentType;
    color?: string;
    theme: typeof Theme;
}

export type { IWrapperProps } from './WrapperTypes';

export interface ICheckBoxProps {
    onClick?: () => void;
    text?: string;
}
