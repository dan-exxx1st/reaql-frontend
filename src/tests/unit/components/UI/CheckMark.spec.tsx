import React from 'react';
import { CheckMark } from 'components/UI';
import { mountWithTheme } from 'tests/helpers/withTheme';

describe('<CheckMark />', () => {
    describe('Snapshots', () => {
        it('Should render with inactive marks and isDouble', () => {
            const wrapper = mountWithTheme(
                <CheckMark isChecked={false} isDouble={true} />
            );

            expect(wrapper).toHaveStyleRule('fill', '#A5A5A5');
            expect(wrapper).toMatchSnapshot();
        });

        it('Should render with active marks and isDouble', () => {
            const wrapper = mountWithTheme(
                <CheckMark isChecked={true} isDouble={true} />
            );
            expect(wrapper).toHaveStyleRule('fill', '#00ADFF');
            expect(wrapper).toMatchSnapshot();
        });

        it('Should render with inactive marks', () => {
            const wrapper = mountWithTheme(
                <CheckMark isChecked={false} isDouble={false} />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('Should render with active marks', () => {
            const wrapper = mountWithTheme(
                <CheckMark isChecked={true} isDouble={false} />
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
