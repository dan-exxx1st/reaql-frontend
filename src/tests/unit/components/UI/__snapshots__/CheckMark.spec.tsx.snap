// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckMark /> Snapshots Should render with active marks 1`] = `
.c0 {
  fill: #00ADFF;
  width: 20px;
  height: 13px;
}

<svg
  class="c0"
>
  <path
    d="M5.67738 10.2586L14.7785 0L16 1.37556L5.67738 13L0.868774 7.58495L2.09888 6.21908L5.67738 10.2586Z"
  />
</svg>
`;

exports[`<CheckMark /> Snapshots Should render with active marks and isDouble 1`] = `
.c0 {
  fill: #00ADFF;
  width: 20px;
  height: 13px;
}

<svg
  class="c0"
>
  <path
    d="M0 7.58495L4.8086 13L6.02151 11.6244L1.22151 6.21908L0 7.58495ZM18.7785 0L9.67742 10.2586L6.09893 6.21908L4.86882 7.58495L9.67742 13L20 1.37556L18.7785 0ZM15.1312 1.37556L13.9183 0L8.45591 6.15127L9.67742 7.51714L15.1312 1.37556Z"
  />
</svg>
`;

exports[`<CheckMark /> Snapshots Should render with inactive marks 1`] = `
.c0 {
  fill: #A5A5A5;
  width: 20px;
  height: 13px;
}

<svg
  class="c0"
>
  <path
    d="M5.67738 10.2586L14.7785 0L16 1.37556L5.67738 13L0.868774 7.58495L2.09888 6.21908L5.67738 10.2586Z"
  />
</svg>
`;

exports[`<CheckMark /> Snapshots Should render with inactive marks and isDouble 1`] = `
.c0 {
  fill: #A5A5A5;
  width: 20px;
  height: 13px;
}

<svg
  class="c0"
>
  <path
    d="M0 7.58495L4.8086 13L6.02151 11.6244L1.22151 6.21908L0 7.58495ZM18.7785 0L9.67742 10.2586L6.09893 6.21908L4.86882 7.58495L9.67742 13L20 1.37556L18.7785 0ZM15.1312 1.37556L13.9183 0L8.45591 6.15127L9.67742 7.51714L15.1312 1.37556Z"
  />
</svg>
`;
